masterplan_tools/method/publicspace/ps_greedy.py

```diff
@@ -0,0 +1,612 @@
+                        if dist["time"] == 20:
+                            to_remove.append(house)
+                    for r in to_remove:
+                        g.remove_edge(key, r)
+
+            # Подсчет количества людей для каждой зоны
+            for node_id in g.nodes():
+                if g.nodes[node_id]["is_zone"] == True:
+                    people = 0
+                    neighbours = g[node_id]
+                    for key, value in neighbours.items():
+                        people += g.nodes[key]["people"]
+                    g.nodes[node_id]["people_around"] = people
+            self.is_graph_exist = True
+
+            # Удаление необеспеченных домов
+            drop = []
+            for n in g:
+                neighbors = list(g.neighbors(n))
+                if len(neighbors) == 0:
+                    drop.append(n)
+            g.remove_nodes_from(drop)
+
+            for n in g:
+                if g.nodes[n]["is_zone"] == True:
+                    neighbor = g.neighbors(n)
+                    if g.nodes[n]["type"] == "существующее":
+                        self.existed.append(n)
+                        if g.nodes[n]["functional_type"] == "Событийное":
+                            for neigh in neighbor:
+                                g.nodes[neigh]["provision_ev"] += 1
+                        if g.nodes[n]["functional_type"] == "Многофункциональное":
+                            for neigh in neighbor:
+                                g.nodes[neigh]["provision_mf"] += 1
+                        if g.nodes[n]["functional_type"] == "Учебное":
+                            for neigh in neighbor:
+                                g.nodes[neigh]["provision_ed"] += 1
+                        if g.nodes[n]["functional_type"] == "Спортивное":
+                            for neigh in neighbor:
+                                g.nodes[neigh]["provision_sp"] += 1
+            self.graph = g
+            return self.graph
+
+        else:
+            return self.graph
+
+    def OPM_info(self):
+        all = 0
+        ev = 0
+        sp = 0
+        mf = 0
+        ed = 0
+        for n in self.graph:
+            if self.graph.nodes[n]["is_zone"] == True and self.graph.nodes[n]["type"] == "существующее":
+                all += 1
+                if self.graph.nodes[n]["functional_type"] == "Событийное":
+                    ev += 1
+                if self.graph.nodes[n]["functional_type"] == "Спортивное":
+                    sp += 1
+                if self.graph.nodes[n]["functional_type"] == "Многофункциональное":
+                    mf += 1
+                if self.graph.nodes[n]["functional_type"] == "Учебное":
+                    ed += 1
+        print(f"количество Многофункциональных ОП: {mf}")
+        print(f"количество Событийных ОП: {ev}")
+        print(f"количество Спортивных ОП: {sp}")
+        print(f"количество Учебных ОП: {ed}")
+        print(f"общее количество ОП: {all}")

```

Comment:
It would be better to return some information structure instead of printing stats right to console.
Same with `provision_info`